lib/cog/commands/relay/list.ex

```diff
@@ -0,0 +1,72 @@
+defmodule Cog.Commands.Relay.List do
+  alias Cog.Commands.Helpers
+  alias Cog.Repository.Relays
+
+  @moduledoc """
+  Lists relays.
+
+  Usage:
+  relay list [-g <group>] [-v <verbose>]
+
+  Flags:
+  -g, --group     Group relays by relay group
+  -v, --verbose   Include additional relay details
+  """
+
+  @doc """
+  Lists relays. Accepts a cog request and returns either a success tuple
+  containing a template and data, or an error.
+  """
+  @spec list_relays(%Cog.Command.Request{}) :: {:ok, String.t, Map.t} | {:error, any()}
+  def list_relays(req) do
+    if Helpers.flag?(req.options, "help") do
+      show_usage
+    else
+      results = case Relays.all do
+        [] ->
+          []
+        relays ->
+          generate_response(req.options, relays)
+      end
+
+      {:ok, "relay-list", results}
+    end
+  end
+
+  defp generate_response(options, relays) do
+    Enum.map(relays, fn(relay) ->
+      relay_map = if Helpers.flag?(options, "verbose") do
+        verbose_relay(relay)
+      else
+        standard_relay(relay)
+      end
+
+      if Helpers.flag?(options, "group") do
+        Map.put(relay_map, "relay_groups", Enum.map(relay.groups, &generate_group_map/1))
+        |> Map.put("_show_groups", true)
+      else
+        relay_map
+      end
+    end)
+  end
+
+  defp verbose_relay(relay) do
```

Comment:
So I actually thought about that @kevsmith, and originally that's how I planned on doing it. But there isn't a way to easily get additional metadata into the template without polluting the record. I could just stick flags in each object to turn things on and off, that's actually what I had to do with the relay groups bit to get it to display properly, but that seems like a generally bad idea. Do you have any thoughts?